### DISCUSSION TITLE
How can I migrate from legacy Prisma to non-legacy Prisma

### DISCUSSION THREAD

**Discussion by leoyli-headsup** [nextauthjs/next-auth#2958] [Help]
So I'm currently using the built-in prisma which is considered as legacy. What I want to know is how can we migrate to v4 in this case? v4 seems don't allow us to map our existing schema. Or I have to write a custom adaptor in this case? Tx!

**Comment by balazsorban44** [comment]
v4 is in fact much more flexible, but more simple at the same time. 

https://next-auth.js.org/getting-started/upgrade-v4

just rename the required fields shown in the docs above, anything else should just work™.

Tell me how it goes.

**Reply by leoyli-headsup** [reply]
@balazsorban44, thanks for the answer, I have read the doc for couple times in case I missed. But the doc don't tell you how can I change the name of the table. We want to change the name of table because we want to align the name with our business. In the contexts of our business, the "account" is referring to a company, while the `user` is a seat that account is paying for. So we want to rename the `account` table as `authentication` or `credential`, something like these. Can I do that now in v4? We are sticking with the legacy adaptor because of that... Or I have write my own custom adaptor if I want to upgrade to v4???

Also, right now in v3 we're using Prisma legacy, which comes with `id` as `int`, so I guess we need to migrate our table to have the `id` in `string`, and add/remove some of the field manually right?

**Reply by balazsorban44** [reply]
Renaming tables with Prisma seem to be as easy as the following:
```js
const prisma = new PrismaClient()
...
adapter: PrismaAdapter({
  ...prisma,
  account: prisma.authentication
}),
```

(I haven't tested it, so the spread operator might not work, but you hopefully get the idea.)

`id`s are expected to be strings (something like UUID) yes, to be less predictable and also match the other adapters.
